### PR TITLE
Add Toggle Rich Presence #20

### PR DESCRIPTION
**Pull request by josago97**
Enabled
image

Disabled
image

### CODE REVIEW

File: VisualStudioDiscordRPC.Shared/PackageController.cs [hunk 1]
```diff
@@ -61,6 +61,7 @@ public PackageController(DTE2 instance, string installationPath)
                 Dte = _instance,
                 ExtensionAssets = extensionsAssetMap,
 
+                Enabled = bool.TryParse(Settings.Default.RichTextEnabled, out bool richTexEnabled) && richTexEnabled,
```
Comment (Ryavell) [inline VisualStudioDiscordRPC.Shared/PackageController.cs:64]: Here is a typo in richTextEnabled.

File: VisualStudioDiscordRPC.Shared/RichPresenceWrapper.cs [hunk 2]
```diff
@@ -287,25 +293,32 @@ private string GetAssetText(Icon icon)
...
+                _presence.Assets.LargeImageText = GetAssetText(largeIcon);
+                _presence.Assets.SmallImageText = GetAssetText(smallIcon);
 
-            _client.SetPresence(_presence);
+                _client.SetPresence(_presence);
+            }
+            else
+            {
+                _client.ClearPresence();
+            }
```
Comment (Ryavell) [inline VisualStudioDiscordRPC.Shared/RichPresenceWrapper.cs:321]: You can invert if to reduce nesting.
```cs
if (!Enabled)
{
    _client.CLearPresence();
    return;
}
// ...
_client.SetPresence(_presence);
```

**Review by Ryavell — CHANGES_REQUESTED** [review]
Thanks for the contribution. There are a couple of places where you need to fix a little.

File: VisualStudioDiscordRPC.Shared/PackageController.cs (hunk 1 above)
Comment (josago97) [reply VisualStudioDiscordRPC.Shared/PackageController.cs:64]: You are right, it should be richPresenceEnabled

File: VisualStudioDiscordRPC.Shared/RichPresenceWrapper.cs (hunk 2 above)
Comment (josago97) [reply VisualStudioDiscordRPC.Shared/RichPresenceWrapper.cs:321]: Okay, I will change it

**Review by Ryavell — APPROVED** [review]

**Comment by Ryavell** [comment]
LGTM :)

**Pull request merged**